File: 121-best-time-to-buy-and-sell-stock/121-best-time-to-buy-and-sell-stock.py
class Solution:
    def maxProfit(self, prices: List[int]) -> int:
        mx = 0
        mini = prices[0]
        n = len(prices)
        for i in range(0,n):
            
            mini = min(mini,prices[i])
            profit = prices[i] - mini
            mx = max(mx,profit)
        return mx
        

File: 121-best-time-to-buy-and-sell-stock/121-best-time-to-buy-and-sell-stock.cpp
class Solution {
public:
    int maxProfit(vector<int>& prices) {
        int lsf = 1e6;
        int op = 0;
        int tp = 0;
        
        for(int i=0;i<prices.size();i++){
            if(prices[i] < lsf){
                lsf = prices[i];
            }
            
            tp = prices[i]-lsf;
            if(op<tp){
                op = tp;
            }
        }
        return op;
    }
};